src/client/v2/serviceClient.ts

```diff
@@ -19,6 +19,9 @@ function convertTokenStringToTokenHeader(
   headerIdentifier: TokenHeaderIdentifier
 ): TokenHeader {
   const tokenHeader = {};
+  if (token === '') {
```

Comment:
Is it clear that in the issue being encountered the empty token provided is of type `string`? I see other possibilities such as `AlgodTokenHeader` so if these were being provided empty, then the fix wouldn't apply.

Reply:
NVM. I see now that we can only get here in the case of `string`: https://github.com/algorand/js-algorand-sdk/blob/5f8097fe2bfcb6b61158d8702c28ad43ca79742c/src/client/v2/serviceClient.ts#L55